### PR TITLE
fix: onlyShowFirstAvailableSlot is enabled, but could still bookable via API or by changing the params

### LINKED ISSUE #16030: [CAL-4098] Only show first slot of day available - other slots still bookable
When enabling 'Only show first slots of each day as available' the slots are shown correctly. But I can still book all other slots via the API or by changing the start time in the URL param on the bookings page. 

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-4098](https://linear.app/calcom/issue/CAL-4098/only-show-first-slot-of-day-available-other-slots-still-bookable)</sub>

### PR DESCRIPTION
**Pull request by RajdeepDs**
## What does this PR do?

This PR fixes an issue where `onlyShowFirstAvailableSlot` was enabled, but could still bookable via the API or by modifying URL params.

- Fixes #16030
- Fixes [CAL-4098](https://linear.app/calcom/issue/CAL-4098/only-show-first-slot-of-day-available-other-slots-still-bookable)

## Updates since last revision

Added server-side validation to prevent bypassing the `onlyShowFirstAvailableSlot` restriction via API or URL manipulation:

- **New validation function** (`validateOnlyShowFirstAvailableSlot.ts`): Validates that the requested booking time is the first available slot of the day when `onlyShowFirstAvailableSlot` is enabled
- **Added `onlyShowFirstAvailableSlot` to DB select** (`getEventTypesFromDB.ts`): Makes the field available in the booking creation flow
- **Integrated validation into booking service** (`RegularBookingService.ts`): Calls validation after `ensureAvailableUsers` populates availability data
- **New error code** (`errorCodes.ts`): Added `BookingNotFirstAvailableSlot` for the validation error

## Visual Demo (For contributors especially)
https://www.loom.com/share/4325894859664c3084edf6912d88ad80?sid=724984fd-f2e8-48fb-8a31-7208f7f4a253

## Human Review Checklist

- [ ] Verify the validation logic correctly identifies the first available slot by comparing with `oooExcludedDateRanges`
- [ ] Check timezone handling edge cases (validation uses event timezone with UTC fallback)
- [ ] Confirm the validation only runs when `skipAvailabilityCheck` is false (intentional for admin/system use)
- [ ] Consider if dedicated unit tests should be added for `validateOnlyShowFirstAvailableSlot`

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

### CODE REVIEW

File: packages/features/bookings/Booker/Booker.tsx [hunk 1]
```diff
@@ -191,15 +194,18 @@ const BookerComponent = ({
     return setBookerState("booking");
   }, [event, selectedDate, selectedTimeslot, setBookerState, skipConfirmStep, layout, isInstantMeeting]);
 
-  const unavailableTimeSlots = isQuickAvailabilityCheckFeatureEnabled
-    ? allSelectedTimeslots.filter((slot) => {
+  const unavailableTimeSlots = !isQuickAvailabilityCheckFeatureEnabled
+    ? []
+    : allSelectedTimeslots.filter((slot) => {
         return !isTimeSlotAvailable({
```
Comment (TusharBhatt1) [inline packages/features/bookings/Booker/Booker.tsx:200]: `isQuickAvailabilityCheckFeatureEnabled ? :` is ideal and more readable

**Review by TusharBhatt1 — COMMENTED** [review]
In this case , the message could be more specific and accurate than : 
image

@CarinaWolli can you confirm

**Comment by kart1ka** [comment]
Hey @RajdeepDs, Could you please resolve merge conflicts?

**Comment by RajdeepDs** [comment]
> Hey @RajdeepDs, Could you please resolve merge conflicts?

Yes, I'll resolve the merge conflicts.

File: packages/trpc/server/routers/viewer/slots/isAvailable.handler.ts [hunk 2]
```diff
@@ -78,6 +79,62 @@ export const isAvailableHandler = async ({
...
+      const prevDate = currentDate.subtract(1, "day").format("YYYY-MM-DD");
+      const nextDate = currentDate.add(1, "day").format("YYYY-MM-DD");
+
+      // Get all slots for the current day and adjacent days
+      const allRelevantSlots = [
+        ...(slotsByDay[prevDate] || []),
+        ...(slotsByDay[date] || []),
+        ...(slotsByDay[nextDate] || []),
+      ];
+
```
Comment (kart1ka) [inline packages/trpc/server/routers/viewer/slots/isAvailable.handler.ts:112]: what is the point of taking available slots from previous and next day into account?

**Review by kart1ka — COMMENTED** [review]
left a comment.

File: packages/trpc/server/routers/viewer/slots/isAvailable.handler.ts (hunk 2 above)
Comment (RajdeepDs) [reply packages/trpc/server/routers/viewer/slots/isAvailable.handler.ts:112]: This code was added because looking up slots only under the current date in GMT can miss valid first-of-day slots for users in other time zones. By including slots from the previous and next days, we ensure that all possible first-available slots are considered, regardless of how they are grouped by date in GMT

https://github.com/calcom/cal.com/pull/21443#discussion_r2100799416
